src/System.CommandLine/CommandLineConfiguration.cs

```diff
@@ -38,6 +38,17 @@ public class CommandLineConfiguration
 
             foreach (var symbol in symbols)
             {
+                foreach (var childSymbol in ((ISymbolSet)symbol.Children).FlattenBreadthFirst(o => o.Children))
+                {
+                    if (childSymbol.Argument.Arity.MaximumNumberOfArguments != 0)
+                    {
+                        if (string.IsNullOrEmpty(childSymbol.Argument.Name))
+                        {
+                            throw new ArgumentException("Name must be set for command arguments.");
```

Comment:
This would be extra helpful if it specified which command's argument needs a name.